MailCore/Utils/Model/Realm/BackgroundRealm.swift

```diff
@@ -38,16 +39,12 @@ public final class BackgroundRealm {
     }
 
     public func execute<T>(_ block: @escaping (Realm) -> T, completion: @escaping (T) -> Void) {
-        BackgroundExecutor.executeWithBackgroundTask { [weak self] taskCompleted in
-            self?.queue.async {
-                guard let realm = self?.getRealm() else { return }
-                completion(block(realm))
-                taskCompleted()
-            }
-        } onExpired: {
-            let expiredBreadcrumb = Breadcrumb(level: .warning, category: "BackgroundRealm")
-            expiredBreadcrumb.message = "Task expired before completing"
-            SentrySDK.addBreadcrumb(expiredBreadcrumb)
+        let expiringActivity = ExpiringActivity()
+        expiringActivity.start()
+        queue.async {
+            let realm = self.getRealm()
```

Comment:
Maybe something like 
```suggestion
        queue.async { [weak self] in
            guard let self else {
                 expiringActivity.endAll()
                 return
            }
            let realm = self.getRealm()
```

Reply:
Yes it is closer to the previous code tu use weak here.

This class goes away in the next PR so I would say, that's not an issue anyway ?

Reply:
As this class no longer exists in the next PR I'm resolving the conversation